modalc-lib/modal-vector.rkt

```diff
@@ -0,0 +1,152 @@
+#lang racket
+
+(require modalc
+         syntax/parse/define)
+
+;; apply mode to both ref and set
+(define (modal-vector/c should-apply-ctc? . inner-ctc)
+  ;; passes inner-ctc in as a list
+  (modal-vector/c* inner-ctc #:ref/set should-apply-ctc?))
+
+
+(define (modal-vector/c* #:ref [ref-ctc? #f]
+                         #:set [set-ctc? #f]
+                         #:ref/set [all-ctc? #f]
+                         . inner-ctc)
+  
+  ;; when using modal-vector/c
+  (when (list? (first inner-ctc))
+    (set! inner-ctc (first inner-ctc)))
```

Comment:
`apply` is what you want (on line 9), rather than special casing the argument shape here on line 18